src/win/conpty.cc

```diff
@@ -8,6 +8,9 @@
  *   with pseudo-terminal file descriptors.
  */
 
+// node versions lower than 10 define this as 0x502 which disables many of the definitions needed to compile
+#define _WIN32_WINNT 0x600
```

Comment:
I think a better version would be to conditionally define it for node<=8

```
diff --git a/binding.gyp b/binding.gyp
index 0c6e3b0..26f8bdf 100644
--- a/binding.gyp
+++ b/binding.gyp
@@ -4,6 +4,13 @@
       'targets': [
         {
           'target_name': 'conpty',
+          'conditions': [
+            ['node_module_version<=57', {
+              'defines': [
+                '_WIN32_WINNT=0x0600',
+              ],
+            }]
+          ],
           'include_dirs' : [
             '<!(node -e "require(\'nan\')")'
           ],
```

Reply:
@deepak1556 do you know if there is a condition that reflects the node module version of the target node? it seems that `node_module_version` only reflects the version of the installed node, not the version that is being compiled against.

Reply:
@dgriffen isn't the version it's compiled against is the only target version it will be able to run under since the modules uses nan/V8 APIs?

Reply:
It seems that https://github.com/nodejs/node-gyp/issues/1004 indicates that there is a bug preventing this from being used correctly.

Reply:
@Tyriar yes, but you can set the `npm_config_target` environment variable to enable compiling against a different version of node.

Reply:
@deepak1556 @Tyriar I think I found an acceptable compromise. `node_module_version` isn't correctly defined for use in the `.gyp` file, but `NODE_MODULE_VERSION` is correctly defined at build time. I added a conditional define in `conpty.cc`.